### ISSUE TITLE
Unexpected tool name 'previewsave'

### ISSUE THREAD

**Issue by brenterprises** [dubreuia/visual_midi#1]
When running this bit of code from your chapter 2 notebook: 
```
from visual_midi import Plotter

# Write the resulting midi file to the output directory
midi_file = os.path.join("output", "out.mid")
mm.midi_io.note_sequence_to_midi_file(sequence, midi_file)
print("Generated midi file: " + str(os.path.abspath(midi_file)))

# Write the resulting plot file to the output directory
plot_file = os.path.join("output", "out.html")
print("Generated plot file: " + str(os.path.abspath(plot_file)))
pretty_midi = mm.midi_io.note_sequence_to_pretty_midi(sequence)
plotter = Plotter()
plotter.show(pretty_midi, plot_file)
```

I get the following stack: 

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-8-a86b0a44e9cc> in <module>
     11 pretty_midi = mm.midi_io.note_sequence_to_pretty_midi(sequence)
     12 plotter = Plotter()
---> 13 plotter.show(pretty_midi, plot_file)

c:\anaconda3\envs\magenta\lib\site-packages\visual_midi\visual_midi.py in show(self, pm, filepath)
    402       :return: the bokeh plot layout
    403     """
--> 404     plot = self.plot(pm)
    405     if self._live_reload:
    406       html = file_html(plot, CDN)

c:\anaconda3\envs\magenta\lib\site-packages\visual_midi\visual_midi.py in plot(self, pm)
    123     plot = bokeh.plotting.figure(
    124       tools="reset,hover,previewsave,wheel_zoom,pan",
--> 125       toolbar_location=preset.toolbar_location)
    126 
    127     # Setup the hover and the data dict for bokeh,

c:\anaconda3\envs\magenta\lib\site-packages\bokeh\plotting\figure.py in figure(**kwargs)
   1362 
   1363 def figure(**kwargs):
-> 1364     return Figure(**kwargs)
   1365 figure.__doc__ = Figure.__doc__
   1366 

c:\anaconda3\envs\magenta\lib\site-packages\bokeh\plotting\figure.py in __init__(self, *arg, **kw)
    164         process_axis_and_grid(self, opts.y_axis_type, opts.y_axis_location, opts.y_minor_ticks, opts.y_axis_label, self.y_range, 1)
    165 
--> 166         tool_objs, tool_map = process_tools_arg(self, opts.tools, opts.tooltips)
    167         self.add_tools(*tool_objs)
    168         process_active_tools(self.toolbar, tool_map, opts.active_drag, opts.active_inspect, opts.active_scroll, opts.active_tap)

c:\anaconda3\envs\magenta\lib\site-packages\bokeh\plotting\_tools.py in process_tools_arg(plot, tools, tooltips)
    126         list of Tools objects added to plot, map of supplied string names to tools
    127     """
--> 128     tool_objs, tool_map = _resolve_tools(tools)
    129 
    130     repeated_tools = [ str(obj) for obj in _collect_repeated_tools(tool_objs) ]

c:\anaconda3\envs\magenta\lib\site-packages\bokeh\plotting\_tools.py in _resolve_tools(tools)
    166             continue
    167 
--> 168         tool_obj = Tool.from_string(tool)
    169         tool_objs.append(tool_obj)
    170         tool_map[tool] = tool_obj

c:\anaconda3\envs\magenta\lib\site-packages\bokeh\models\tools.py in from_string(cls, name)
    156             if not matches:
    157                 matches, text = known_names, "possible"
--> 158             raise ValueError(f"unexpected tool name '{name}', {text} tools are {nice_join(matches)}")
    159 
    160     @classmethod

ValueError: unexpected tool name 'previewsave', possible tools are pan, xpan, ypan, xwheel_pan, ywheel_pan, wheel_zoom, xwheel_zoom, ywheel_zoom, zoom_in, xzoom_in, yzoom_in, zoom_out, xzoom_out, yzoom_out, click, tap, crosshair, box_select, xbox_select, ybox_select, poly_select, lasso_select, box_zoom, xbox_zoom, ybox_zoom, save, undo, redo, reset, help, box_edit, point_draw, poly_draw, poly_edit or hover
```

**Comment by dubreuia** [comment]
Hey @brenterprises, thanks for letting me know.

Seems like its a version problem, I never saw that error but I think Bokeh plot we're using for plotting the resulting MIDI doesn't recognize a toolbar parameter. As a workaround, remove the `plotter.show` line, so you can continue working (open the MIDI file in another software).

Can you send me your python, visual midi, and bokeh plot version? I'll check how I can circumvent this. Thanks

**Comment by brenterprises** [comment]
Hey @dubreuia , thanks for the quick response. Just started going through your book and it's really cool stuff. I'm using Magenta 1.1.7, visual-midi 0.0.3, and python 3.6.10. I've been working around the plots right now. Might have more questions for you about your book later on if that's OK?

**Comment by dubreuia** [comment]
Thanks I appreciate it :-)

What is your bokeh plot version? In a python terminal your can probably do `import bokeh` and `print(bokeh.__version__)`.

If you have questions about the code, you can post it on the book's github, or send me an email if it is about the book in general.

I'll check you problem soon. Cheers

**Comment by brenterprises** [comment]
bokeh 2.0.1 

Thanks, I think I'll have quite a few questions, both clarifications and some requests for advice. :)

**Comment by dubreuia** [comment]
The `previewsave` tool was removed in 2.0.0: https://github.com/bokeh/bokeh/commit/f033f00549bc7725e5ca2442fc0dfc1621dbec2c#diff-e42bf369acbbcbcedf2ed24df6022bd1L92

Seems like I need to replace it for `save`. I'll fix this and release a new version

**Comment by dubreuia** [comment]
Released in 1.0.0, you can upgrade with `pip install --upgrade visual_midi`
